PMMain/Midas/MidasGUI/MidasThread.h

```diff
@@ -21,7 +25,7 @@ class MidasThread : public QThread
 
     void threadEmitString(std::string str);
     void threadEmitStateString(std::string str);
-
+    std::vector<ringData> readKeyboardSetupFile();
 private:
```

Comment:
As per comment above, this function should be elsewhere (maybe even it's own class "KeyboardSettingsReader" or something)